### PR TITLE
Refactor AppComponent

### PR DESCRIPTION
**Pull request by anmolarora1**

### CODE REVIEW

**Comment by raineorshine** [comment]
I can understand defining the container separately to improve readability. However I'm not sure I agree with having them in separate files. It seems they would also be used in conjunction, and it would be more manageable to have them co-housed in the same file.

**Comment by anmolarora1** [comment]
> I can understand defining the container separately to improve readability. However I'm not sure I agree with having them in separate files. It seems they would also be used in conjunction, and it would be more manageable to have them co-housed in the same file.

I agree that they would be used in the same file. However, I think it makes sense to have them in separate files because
1. Promotes readability - Because the files are shorter
2. Separation of concerns and responsibilities - The container handles the responsibility of supplying inputs to the component in the desired format (with or without transformations)whereas the component deals with presentational logic without the need to know where and how the input is coming from.
3. Tests - We can test each file as a function and testing becomes relatively easier and more understandable
A solution could be to keep the 'component' and 'container' files in a single folder. 
It's just an opinion based on my experience. If you and other devs think otherwise, I'll merge them into a single file and raise a new PR.

File: src/components/AppComponent/AppComponent.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+          {isTutorial() && !isLoading ? <Tutorial /> : null}
+
+          <SplitPane
+            style={{ position: 'relative' }}
+            className={isSplitting ? 'animating' : ''}
+            split="vertical"
+            defaultSize={!splitView ? '100%' : parseInt(localStorage.getItem('splitPos'), 10) || '50%'}
+            size={!splitView ? '100%' : parseInt(localStorage.getItem('splitPos'), 10) || '50%'}
+            onChange={size => localStorage.setItem('splitPos', size)}>
```
Comment (raineorshine) [inline src/components/AppComponent/AppComponent.js]: These should be memoized

File: src/containers/AppComponent/AppComponent.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  meta,
+} from '../../util'
+import { AppComponent } from '../../components/AppComponent/AppComponent'
+
+const mapStateToProps = state => {
+  const { dataNonce, focus, search, user, dragInProgress, isLoading, showModal, showSplitView } = state
+  const dark = (isLoading ? localStorage['Settings/Theme'] : getSetting('Theme')[0]) !== 'Light'
+  const scaleSize = (isLoading ? +(localStorage['Settings/Font Size'] || 16) : getSetting('Font Size')[0] || 16) / 16
+  const tutorial = isLoading ? localStorage['Settings/Tutorial'] === 'On' : meta([EM_TOKEN, 'Settings', 'Tutorial']).On
+  const tutorialStep = isLoading ? +(localStorage['Settings/Tutorial Step'] || 1) : getSetting('Tutorial Step')[0] || 1
```
Comment (raineorshine) [inline src/containers/AppComponent/AppComponent.js]: These should be memoized

**Pull request merged**